crates/avalanche/src/upload.rs

```diff
@@ -36,11 +36,14 @@ pub async fn upload(state: State, build: BuilderFinished, token: String, uri: &s
 
     let (sender, receiver) = mpsc::channel(1);
 
-    tokio::spawn(async move {
-        if let Err(e) = client.upload(ReceiverStream::new(receiver)).await {
-            error!(error = error::chain(e), "Upload stream failed");
+    tokio::spawn(
+        async move {
+            if let Err(e) = client.upload(ReceiverStream::new(receiver)).await {
+                error!(error = error::chain(e), "Upload stream failed");
+            }
         }
-    });
+        .instrument(Span::current()),
```

Comment:
```suggestion
        .in_current_span(),
```

https://docs.rs/tracing/latest/tracing/trait.Instrument.html#method.in_current_span

Reply:
Oh nice, TIL